### PR TITLE
Fix ip per container test

### PR DESCRIPTION
**Pull request by sargun**

### CODE REVIEW

**Comment by sargun** [comment]
@cmaloney, or @alberts Can we run tests multiple time with this branch, and look at success rate?

**Comment by cmaloney** [comment]
I'll launch 15 of the vagrant, see how that goes.

**Comment by cmaloney** [comment]
Much less flaky, although still at least one failure: https://teamcity.mesosphere.io/viewLog.html?buildId=356042&tab=buildResultsDiv&buildTypeId=DcosIo_Dcos_VagrantIntegrationTests_ProvisioningTestGenconfBashPr

File: packages/dcos-integration-test/extra/test_networking.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    """
+    # Launch the test_server in ip-per-container mode
+
+    app_definition, test_uuid = cluster.get_base_testapp_definition(ip_per_container=True)
+
+    app_definition['instances'] = 2
+    if len(cluster.slaves) >= 2:
+        app_definition['constraints'] = [['hostname', 'UNIQUE']]
+    else:
+        logging.warning('The IP Per Container tests needs 2 (private) agents to work')
```
Comment (cmaloney) [inline packages/dcos-integration-test/extra/test_networking.py]: This should be a logging.error() then hard fail. Running the integration tests with less than two private agents isn't supported.

**Comment by cmaloney** [comment]
https://teamcity.mesosphere.io/viewLog.html?buildId=356067&buildTypeId=DcosIo_Dcos_VagrantIntegrationTests_ProvisioningTestGenconfBashPr&tab=buildResultsDiv

**Comment by cmaloney** [comment]
And a spartan watchdog failure: https://teamcity.mesosphere.io/viewLog.html?buildId=356174&tab=buildResultsDiv&buildTypeId=DcosIo_Dcos_VagrantIntegrationTests_ProvisioningTestGenconfBashPr

File: packages/dcos-integration-test/extra/test_networking.py (hunk 1 above)
Comment (sargun) [inline packages/dcos-integration-test/extra/test_networking.py]: Was this always the case? When I first wrote this test, I found clusters with only one public, and one private.

**Comment by cmaloney** [comment]
https://teamcity.mesosphere.io/viewLog.html?buildId=356182&buildTypeId=DcosIo_Dcos_VagrantIntegrationTests_ProvisioningTestGenconfBashPr&guest=1

File: packages/dcos-integration-test/extra/test_networking.py (hunk 1 above)
Comment (cmaloney) [inline packages/dcos-integration-test/extra/test_networking.py]: A long time ago there was a single node vagrant, for quite a while there should always be minimum 2 private agents.

Comment (sargun) [inline packages/dcos-integration-test/extra/test_networking.py]: I'll change this in a follow-on commit.

**Comment by cmaloney** [comment]
Muted test_applications.test_if_minuteman_routes_to_named_vip on the vagrant bash, was causing quite a few other prs to fail

**Comment by spahl** [comment]
This does not fix all the flakyness but introduces a lot of bugfixes (see commits). :ship:

**Pull request merged**